integrators/symmetric_linear_multistep_integrator_body.hpp

```diff
@@ -0,0 +1,500 @@
+﻿#pragma once
+
+#include "integrators/symmetric_linear_multistep_integrator.hpp"
+
+#include <algorithm>
+#include <vector>
+
+#include "integrators/symplectic_runge_kutta_nyström_integrator.hpp"
+
+namespace principia {
+namespace integrators {
+
+namespace {
+
+// For the Adams-Moulton integrators, see
+// http://www.scholarpedia.org/article/Linear_multistep_method#Methods_for_Hamiltonian_systems  // NOLINT
+// and
+// https://en.wikipedia.org/wiki/Linear_multistep_method#Adams.E2.80.93Moulton_methods.  // NOLINT
+// The latter has a general formula for the coefficients and was used to
+// compute the high-order integrators.
+
+AdamsMoulton<1> const& AdamsMoultonOrder1() {
```

Comment:
There is a unique Adams-Moulton method of order `k`. Make this a function template and write specializations. Moreover, I'd like `AdamsMoulton` to be in a separate file: even though we do not implement the actual Adams-Moulton method, it is a different thing which we happen to reuse for the symmetric linear multistep integrator.